cmd/nginx-ingress/main_test.go

```diff
@@ -22,3 +23,77 @@ func TestValidateStatusPort(t *testing.T) {
 	}
 
 }
+
+func TestParseNginxStatusAllowCIDRs(t *testing.T) {
+
+	var tests = []struct {
```

Comment:
I suggest to further simplify this test. You can have two tests slices:
```
	var testsWithValidInput = []struct {
		input    string
		expected []string
	}{
		// ...
	}

	var testsWithInvalidInput = []struct {
		input string
	}{
		// ...
	}
```
I don't think it is necessary to check the error messages. Checking the fact that an error is indeed returned is enough. 

To compare expected input and result, you can do the following:
```
		if !reflect.DeepEqual(test.expected, splitArray) {
			t.Errorf("parseNginxStatusAllowCIDRs(%q) expected %q returned %q", test.input, test.expected, splitArray)
		}
```
Note that reflect.DeepEqual will only be called for `testsWithValidInput` tests.